### DISCUSSION TITLE
static-delegates: Obsolote?

### DISCUSSION THREAD

**Discussion by bernd5** [dotnet/csharplang#4445] [General] [answered]
Isn't this proposal now obsolote with the function-pointer feature introduced in CSharp 9?

In my eyes: yes.

**Comment by 333fred** [comment]
Could you reference the issue you're talking about?

**Comment by huoyaoyuan** [comment, marked as answer]
There was a feature proposal called static delegate.
In 9.0, the similar functionality has been supported by function pointers. Static delegate proposal should be *replaced by alternative*.

**Reply by 333fred** [reply]
There was a feature proposal... where?

**Reply by BreyerW** [reply]
Found it [Static delegate](https://github.com/dotnet/csharplang/issues/302)

**Reply by bernd5** [reply]
The paper can be found [here](https://github.com/dotnet/csharplang/blob/master/proposals/static-delegates.md)

**Reply by huoyaoyuan** [reply]
https://github.com/dotnet/csharplang/issues/302#issuecomment-630282294 Fred had answered it

**Reply by bernd5** [reply]
Function pointers are shipped now ...

**Reply by Joe4evr** [reply]
![](https://i.imgur.com/jks6Qqr.png)

**Reply by bernd5** [reply]
It should be deleted / marked as obsolote / closed as written: https://github.com/dotnet/csharplang/issues/302#issuecomment-693003548

**Reply by 333fred** [reply]
Thanks all, issues are closed and the spec is moved to rejected.
